domain/src/gateway/tiptap.rs

```diff
@@ -33,3 +34,81 @@ async fn build_auth_headers(config: &Config) -> Result<reqwest::header::HeaderMa
     headers.insert(reqwest::header::AUTHORIZATION, auth_value);
     Ok(headers)
 }
+
+pub struct TiptapDocument {
+    client: reqwest::Client,
+    base_url: String,
+}
+
+impl TiptapDocument {
+    pub async fn new(config: &Config) -> Result<Self, Error> {
+        let client = client(config).await?;
+        let base_url = config.tiptap_url().ok_or_else(|| {
+            warn!("Failed to get Tiptap URL from config");
+            Error {
+                source: None,
+                error_kind: DomainErrorKind::Internal(InternalErrorKind::Other),
+            }
+        })?;
+        Ok(Self { client, base_url })
+    }
+
+    pub async fn create(&self, document_name: &str) -> Result<(), Error> {
+        let url = format!(
```

Comment:
I suggest pulling this up to the top of this module since it's repeated for `create` and `delete`, which will also help centralize it in case TipTap ever changes their cloud API paths.

Reply:
Great idea 3e0a566cda9b49ef57e216f199d3b6faa75d8020